packages/shared/src/hooks/utils/useTrackedLink.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useGetShortUrl } from './useGetShortUrl';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { ReferralCampaignKey } from '../../lib/referral';
+import { disabledRefetch } from '../../lib/func';
+
+interface UseTrackedLinkProps {
+  enabled?: boolean;
+  link: string;
+  cid: ReferralCampaignKey;
+}
+
+interface UseTrackedLink {
+  shareLink: string;
+  isLoading: boolean;
+}
+
+export const useTrackedLink = ({
+  enabled = true,
+  link,
+  cid,
+}: UseTrackedLinkProps): UseTrackedLink => {
```

Comment:
This is another hook we introduce for links, feels like it is getting out of hand 😆 

I think we should be able to somehow manage it from `useGetShortUrl`and implement the logic specific to this one component.

Also that hook already stores short url under `RequestKey.ShortUrl` key so we for sure don't need `TrackingLink`. 

Personally I think just having logic

Reply:
Yep, I don't even know how much hook we have that processes copying links 😅

Reply:
Makes sense. There is actually quite a bit to refactor here. For example, this function could have been in a `useMutation` hook than `queryClient.fetchQuery` so we can track loading phase.

I would love to refactor everything here now, but can we maybe create another ticket instead just so we can merge this as they are considering this issue as something critical that we should fix asap. 

![Screenshot 2024-03-15 at 9 36 28 PM](https://github.com/dailydotdev/apps/assets/13744167/ea4f0202-308c-466b-a599-3f21de0bb80b)

Reply:
I can even do an immediate follow up PR after this gets merged.

Reply:
I agree we should keep the footprint small for this, so I think we can just avoid adding any hooks in the first place and implement logic you added in that one component?

Reply:
Just because I am sure by adding yet another hook we will confuse the usage even more 😆

Reply:
Yeah I wouldn't rush to push this out.
Rather fix it properly I think 🤔

Reply:
Yeah, sure, I'd be glad to do the refactoring here, I was just considering the product's urgency on the matter.

Reply:
No need to refactor here, just move the hook logic to that single component you are using it in? To not block you and we can add/talk about refactoring as tech debt. Would that work?

Reply:
Feels like it is the same thing with the number of code changes with the existing hook, which is basically another thing to clean up since the hook is now doing two things and we don't even need the original callback.